core/trino-main/src/main/java/io/trino/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -766,7 +766,7 @@ private Type handleResolvedField(Expression node, ResolvedField resolvedField, C
             }
 
             if (field.getOriginTable().isPresent() && field.getOriginColumnName().isPresent()) {
-                tableColumnReferences.put(field.getOriginTable().get(), field.getOriginColumnName().get());
+                tableColumnReferences.put(field.getOriginTable().get(), field.getOriginColumnName().get().toLowerCase(ENGLISH));
```

Comment:
@martint I sprinkled those `toLowerCase` in enough places to fix this particular issue, but I'm not entirely sure if these are the right places. I think the convention is to keep the original case in the AST and only convert it as needed in the analyzer, even though it would probably be easier to fix all issues like this by storing converted names in AST. WDYT?

Reply:
Identifiers need to be resolved according to standard sql rules. It differs for quoted vs non-quoted identifiers.

Currently, there are some inconsistencies  in how they are being handled in various places, but we have a long standing issue to fix them. See https://github.com/trinodb/trino/issues/17

Reply:
I think we can fix this by associating masks to the corresponding Field in StatementAnalyzer and Analysis, instead of doing it by name. Then, in RelationPlanner, the lookup can be done directly by Field instead of relying on the name having the right case.

Reply:
@martint I pushed a prototype of a solution which (I think) implements your suggestion. There's no way to do efficient lookups of `Field`s, though, so in some places we need to do a linear scan of the list of masks in order to find a match.